src/samples/doxygen/switch_map.cpp

```diff
@@ -0,0 +1,16 @@
+#include <rpp/rpp.hpp>
+#include <iostream>
+
+/**
+ * \example switch_map.cpp
+ **/
+int main()
+{
+    //! [switch_map]
+    rpp::source::just(1, 2, 3)
+            .switch_map([](int val) { return rpp::source::from_iterable(std::vector(val, val)); })
```

Comment:
can you add something with `never` to show this important moment?

Reply:
Sounds good!